Reemit.Common/RangeMapped.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace Reemit.Common;
+
+public class RangeMapped<TValue>(int position, int length, TValue value) : IRangeMapped
```

Comment:
I wonder if using `readonly struct` instead of `class` here could help us avoid some heap allocations in certain circumstances. I benchmarked the following code copied from one of your unit tests:

```csharp
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using Reemit.Common;

var summary = BenchmarkRunner.Run<Bench>();

[MemoryDiagnoser]
public class Bench
{
    [Benchmark]
    public void CreateScope_Called_TracksRange()
    {
        // Arrange
        byte[] bytes = [0x0, 0x1, 0x2, 0x3, 0x4, 0x5];
        using var stream = new MemoryStream(bytes);
        using var binaryReader = new BinaryReader(stream);
        const int sharedReaderOffset = 1;
        using var sharedReader = new SharedReader(sharedReaderOffset, binaryReader, new object());
        const int readBytesCount = 2;
        byte[] actualBytes;
        RangeMapped<byte[]> actualRangeMappedBytes;

        // Act
        using (var scope = sharedReader.CreateRangeScope())
        {
            actualBytes = sharedReader.ReadBytes(readBytesCount);
            actualRangeMappedBytes = scope.ToRangeMapped(actualBytes);
        }
    }
}
```
Results with RangeMapped<TValue> being declared as `struct`:
<img width="867" alt="image" src="https://github.com/v0idzz/Reemit/assets/12448522/8a374a54-28e1-42d8-ae03-2552e0bd616f">

and as `class`:
<img width="793" alt="image" src="https://github.com/v0idzz/Reemit/assets/12448522/dd22e355-4eff-4922-b9c8-320abf2d2eab">

Reply:
Good call, that's a rather significant improvement. Fixed.